src/mono/mono/mini/alias-analysis.c

```diff
@@ -53,10 +53,13 @@ lower_load (MonoCompile *cfg, MonoInst *load, MonoInst *ldaddr)
 	}
 
 	if (replaced_op != load->opcode) {
-		if (cfg->verbose_level > 2)
+		if (cfg->verbose_level > 2) {
+			#ifndef DISABLE_LOGGING
 			printf ("Incompatible load type: expected %s but got %s\n",
```

Comment:
What is the purpose of these changes ? These are in the JIT, and not many users would disable logging and not disable the JIT.

Reply:
Since wrapping the `mono_inst_name` function declaration and functiond definition created behavior such as `mono_inst_name` that isn't protected by `#ifndef DISABLE_LOGGING` into a compile-time error that will fail to build the runtime, I have wrapped `mono_inst_name` function call with `#ifndef DISABLE_LOGGING` so that the android build test doesn't fail. Without changing this in all files, the code was failing in runtime, since On Android, iOS and WASM we build the runtime with DISABLE_LOGGING defined.
https://github.com/dotnet/runtime/issues/83545#issue-1628193099

Reply:
Wouldn't be simpler to change the declaration in mini.h in one place ?

Reply:
> <img alt="Screenshot 2023-09-11 at 13 21 16" width="1891" src="https://user-images.githubusercontent.com/55002338/267004359-b7cf18cd-bdd2-41be-bfd2-72fc28e145a9.png">

I have tried that. my first commit was with only changing the declarating and definition. which resulted to the above CI/CD fail: implicit declaration of the function mono_inst_name. However, wrapping the function call with DISABLE_LOGGING everywhere seemed to fix that implicit declaration of function error.

Reply:
@vargaz do you have a suggestion how this should be fixed?

Reply:
My suggestion would be to change all the call sites to include:
```
#ifndef DISABLE_LOGGING
        // some call site
#else
	g_error ("unknown opcode name for %d", op);
	g_assert_not_reached ();
#endif
```
as we still want to assert and display some error message - but without details for logging.

@lambdageek what do you think?

Reply:
I think the original problem itself is not serious enough to warrant this amount of changes.

Reply:
We are just starting the .NET9 cycle. I think it is fine to proceed with extensive changes. We should be mindful of performance implications if any.

Reply:
> My suggestion would be to change all the call sites to include:
> 
> ```
> #ifndef DISABLE_LOGGING
>         // some call site
> #else
> 	g_error ("unknown opcode name for %d", op);
> 	g_assert_not_reached ();
> #endif
> ```
> 
> as we still want to assert and display some error message - but without details for logging.

should I implement this to see the changes? @ivanpovazan

Reply:
@sayeedkhannabil sorry for the delay, I would suggest implementing the requested changes and get all the CI checks to pass.